tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -233,8 +247,16 @@ def apply_ref(value):
         for key, value in node.items():
             node[key] = apply_ref(value)
     elif isinstance(node, list):
-        for i, value in enumerate(node):
-            node[i] = apply_ref(value)
+        results = []
+        for _, value in enumerate(node):
```

Comment:
Enumerate is now useless
```suggestion
        for value in node:
```